core/templates/pages/contributor-dashboard-page/services/contribution-and-review.service.ts

```diff
@@ -16,158 +16,179 @@
-require('domain/suggestion/SuggestionThreadObjectFactory.ts');
-require('pages/exploration-editor-page/services/exploration-data.service.ts');
-require('services/alerts.service.ts');
-
-require(
-  'pages/exploration-editor-page/exploration-editor-page.constants.ajs.ts');
-
-angular.module('oppia').factory('ContributionAndReviewService', [
-  '$http', 'UrlInterpolationService', 'ACTION_ACCEPT_SUGGESTION',
-  function(
-      $http, UrlInterpolationService, ACTION_ACCEPT_SUGGESTION) {
-    var _SUBMITTED_SUGGESTION_LIST_HANDLER_URL = (
-      '/getsubmittedsuggestions/<target_type>/<suggestion_type>');
-    var _REVIEWABLE_SUGGESTIONS_HANDLER_URL = (
-      '/getreviewablesuggestions/<target_type>/<suggestion_type>');
-    var _SUGGESTION_TO_EXPLORATION_ACTION_HANDLER_URL = (
-      '/suggestionactionhandler/exploration/<exp_id>/<suggestion_id>');
-    var _SUGGESTION_TO_SKILL_ACTION_HANDLER_URL = (
-      '/suggestionactionhandler/skill/<skill_id>/<suggestion_id>');
-    var _UPDATE_TRANSLATION_HANDLER_URL = (
-      '/updatetranslationsuggestionhandler/<suggestion_id>'
-    );
-    var _UPDATE_QUESTION_HANDLER_URL = (
-      '/updatequestionsuggestionhandler/<suggestion_id>'
-    );
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+import { AppConstants } from 'app.constants';
+import { ContributionAndReviewBackendApiService }
+  from './contribution-and-review-backend-api.service';
+import { SuggestionBackendDict } from 'domain/suggestion/suggestion.model';
+import { StateBackendDict } from 'domain/state/StateObjectFactory';
+import { ImagesData } from 'services/image-local-storage.service';
+
+export interface OpportunityDict {
+  'skill_id': string;
+  'skill_description': string;
+}
+
+export interface FetchSuggestionsResponse {
+  'target_id_to_opportunity_dict': {
+    [propName: string]: OpportunityDict;
+  };
+  suggestions: SuggestionBackendDict[];
+}
+
+interface FetchSuggestionsReturn {
+  [propName: string]: {
+    suggestion: SuggestionBackendDict;
+    details: OpportunityDict;
+  };
+}
 
-    var _fetchSuggestionsAsync = async function(url) {
-      return $http.get(url).then(function(res) {
-        var suggestionIdToSuggestions = {};
-        var targetIdToDetails = res.data.target_id_to_opportunity_dict;
-        res.data.suggestions.forEach(function(suggestion) {
+@Injectable({
+  providedIn: 'root',
+})
+export class ContributionAndReviewService {
+  constructor(
+    private carbas: ContributionAndReviewBackendApiService
+  ) {}
+
+  private async _fetchSuggestionsAsync(
+      urlName: string, targetType: string, suggestionType: string
+  ): Promise<FetchSuggestionsReturn> {
+    return (
+      this.carbas.fetchSuggestionsAsync(
+        urlName, targetType, suggestionType
+      ).then((data) => {
+        var suggestionIdToSuggestions: FetchSuggestionsReturn = {};
+        var targetIdToDetails = data.target_id_to_opportunity_dict;
```

Comment:
Always prefer `let/const` over `var`.

Reply:
Forgot to change them previously. Changed them now.